.github/workflows/pull-request.yml

```diff
@@ -41,6 +41,22 @@ jobs:
           # output: ::$ANNOTATION_SEVERITY file=icons/accessibility.json,line=2,endLine=3,title=$ANNOTATION_TITLE::$ANNOTATION_DESCRIPTION%0A%0A+  "contributors": ["hi"],%0A@@ -13 +14 @@%0A+}%0A
       - name: Fail if contributors have changed
         run: git diff --exit-code -- icons/*.json
+
+  lint-icons:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: pnpm/action-setup@v2
+        with:
+          version: 8
+      - uses: actions/setup-node@v3.8.1
+        with:
+          node-version: 18
+          cache: 'pnpm'
+
+      - name: Run ESLint
+        run: pnpm lint
```

Comment:
I would really try to avoid running `pnpm install` on `pull_request_target`, it's slow and not really secure.

Alternatively, we can call `https://lucide.dev/api/format`, write the results to the file do a diff like we do for the contributors.

The `/api/format` route would use [`optimizeSvg`](https://github.com/lucide-icons/lucide/blob/main/scripts/render/processSvg.mjs).

Reply:
I've created a separate workflow with just the `pull_request` trigger. I will now lint all files.